src/server/page_server.cpp

```diff
@@ -30,13 +30,24 @@
 #include "page_buffer.h"
 #include "server_type.hpp"
 #include "system_parameter.h"
+#include "thread_entry_task.hpp"
 #include "vpid_utilities.hpp"
 
 #include <cassert>
 #include <cstring>
 #include <functional>
 #include <thread>
 
+page_server::page_server (const char *db_name)
+  : m_server_name { db_name }
+{
+  const auto thread_count = 4; // Arbitrary chosen
+  const auto task_max_count = thread_count * 4;
+
+  m_workpool = cubthread::get_manager ()->create_worker_pool (thread_count, task_max_count, "page_server_workers",
```

Comment:
Introducing the worker pool, we have three worker pools in the `page_server`, one for the request responder for tran servers, one for the request responder for follower servers, and one for the catch-up.
I think it can be integrated into one and shared, but it's arguable. Please give me your opinions. It affects #4667. See https://github.com/CUBRID/cubrid/pull/4667#discussion_r1319572377.

Reply:
I have a little knowledge about thread manager, so I have a question! 

It seems that one followee is responsible for handling requests from multiple followers. Depending on the situation, it's possible that there are more follower requests than the number of threads available in the worker pool. What happens when there are no more available threads in the worker pool? Does it wait until it can acquire thread resource to process follower requests, or does it mean it can't perform follower requests because it couldn't allocate thread?

If the situation leads to not being able to perform follower requests, would it be a good idea to prevent this by setting a maximum number of followers that can be processed at once using system parameters, for instance?

Additionally, would it be a good approach to use cubthread::get_manager()->is_pool_full() or cubthread::get_manager()->try_task() to inform followers whether their requests can be processed?

Reply:
Oh, I see. Thank you for explaining. It might be a problem if it dies without acquiring the thread resource, but it should be fine if it waits until it gets the thread resource. The rest should really be discussed as part of the optimization phase, as you suggested.